packages/react-native/ReactCommon/react/performance/timeline/PerformanceObserver.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "PerformanceObserver.h"
+#include "PerformanceEntryReporter.h"
+
+namespace facebook::react {
+
+void PerformanceObserver::handleEntry(const PerformanceEntry& entry) {
+  // https://w3c.github.io/performance-timeline/#takerecords-method (step 7.1)
+  // https://www.w3.org/TR/event-timing/#should-add-performanceeventtiming
+  if (entry.entryType == PerformanceEntryType::EVENT) {
+    if (entry.duration < durationThreshold_) {
+      // The entries duration is lower than the desired reporting threshold,
+      // skip
+      return;
+    }
+  }
+
+  if (observedTypes_.contains(entry.entryType)) {
+    buffer_.push_back(entry);
```

Comment:
oops, we're not scheduling the task to notify the observer anywhere :S

Reply:
I need to get those unittests working somehow 🥲

Reply:
Looking at it again, we call the `callback` below, which is passed from `NativePerformanceObserver` to schedule a task 🤔